terraform/modernisation-platform-account/iam.tf

```diff
@@ -372,4 +372,23 @@ resource "aws_iam_policy" "instance-scheduler" {
 resource "aws_iam_role_policy_attachment" "instance-scheduler" {
   policy_arn = aws_iam_policy.instance-scheduler.arn
   role       = aws_iam_role.instance-scheduler.name
-}
\ No newline at end of file
+}
+
+# Create assumable roles with managed policies
+# This is for the github-actions role within the member SSO accounts to be able to assume this githubreadonly
+# role to make the initial OIDC request through the AWS provider
+module "iam_assumable_roles" {
+  source               = "terraform-aws-modules/iam/aws//modules/iam-assumable-roles"
+  version              = "~> 4.11"
+  max_session_duration = 43200
+
+  # Read-only role
+  create_readonly_role       = true
+  readonly_role_name         = "githubreadonly"
+  readonly_role_requires_mfa = true
+
+  # Allow created users to assume these roles
+  trusted_role_arns = [
```

Comment:
This will need to be for all member accounts, so the trust policy might have to have `*` as the principal and a condition similar to:
```
    condition {
      test     = "ForAnyValue:StringLike"
      variable = "aws:PrincipalOrgPaths"
      values   = ["${data.aws_organizations_organization.root_account.id}/*/${local.modernisation_platform_ou_id}/*"]
    }
```

Reply:
I'm not sure if the module supports passing in of policies or conditions? If not it might have to be created using individual resources rather than the module.